test/metabase/api/search_test.clj

```diff
@@ -596,3 +596,11 @@
               (db/update! Pulse (:id pulse) :dashboard_id (:id dashboard))
               (is (= nil
                      (search-for-pulses pulse))))))))))
+
+(deftest non-native-card-dataset-query-test
+  (testing "You shouldn't see non-native Cards in the search results that match the keys of the dataset_query"
+    ;; https://github.com/metabase/metabase/issues/24132
+    (mt/with-temp Card [card {:name "Round Table" :query_type "query" :dataset_query {:query "select * from round_table"}}]
```

Comment:
This test doesn't seem to match up with the description. I would expect it to include a native and a non-native card with a query string that would match both, but only the native card is returned. One example is that you could have one query `"select id, foo from round_table"` and then an mbql card that includes an expression named `"foo"` that is just `(+ 1 1)`, and search for foo.

Reply:
something like
```clojure
(deftest non-native-card-dataset-query-test
  (testing "You shouldn't see non-native Cards in the search results that match the keys of the dataset_query"
    ;; https://github.com/metabase/metabase/issues/24132
    (mt/with-temp Card [native-card {:name "native"
                                     :query_type "query"
                                     :dataset_query {:query "select * from foo_bar"}}
                        mbql-card   {:name "mbql"
                                     :query_type "query"
                                     :dataset_query {:query {:expressions {"foo_bar" [:+ 1 1]}}}}]
      (do-test-users [user [:rasta]]
                     (is (= [native-card]
                            (search-request-data user :q "foo_bar")))))))
```
But I don't see this test matching anything. Ideally it would match both cards _before_ the change, and then you evaluate the new code and it matches only a single card.

Reply:
Good catch, I had some changes to this I forgot to push. I've now updated the test.